lib/puppet/provider/service/base.rb

```diff
@@ -31,8 +31,9 @@ def getps
   def getpid
     @resource.fail "Either stop/status commands or a pattern must be specified" unless @resource[:pattern]
     regex = Regexp.new(@resource[:pattern])
+    ps = getps
     self.debug "Executing '#{ps}'"
-    IO.popen(getps) { |table|
+    IO.popen(ps) { |table|
```

Comment:
It's so easy to miss string interpolation references :/  For a second I saw this and thought why was this change necessary.